### PR TITLE
#168970174 Implement category view api endpoint

### PR DESCRIPTION
**Pull request by sebalu**
#### What does this PR do?
- Implements a category view api endpoint
- Refactors unit tests
- deprecates redundant utility functions
#### Description of Task to be completed?
:heavy_check_mark: Employee can view articles by querying a single category
image

:heavy_check_mark: Refactor unit tests

:heavy_check_mark: Remove duplicate utility functions

#### How should this be manually tested?
- follow the installation guidelines in the Readme.MD
- test the endpoint ```/api/v1/search?category=pivotal-tracker```
#### What are the relevant pivotal tracker stories?
[Finishes #168970174](https://www.pivotaltracker.com/story/show/168970174)
[Finishes #168990177](https://www.pivotaltracker.com/story/show/168990177)
[Finishes #168992166](https://www.pivotaltracker.com/story/show/168992166)

### CODE REVIEW

File: server/controllers/article.controller.js [hunk 1]
```diff
@@ -56,18 +43,13 @@ class Article {
...
 
   static deleteArticle(req, res) {
-    const article = articleModel.findArticle(parseInt(req.params.id, 10));
-    if (article === undefined) {
-      return res.status(404).json({
-        message: `An article with the unique id: ${req.params.id} does not exist`
-      });
-    }
     articleModel.deleteArticle(parseInt(req.params.id, 10));
     return res.status(200).json({
```
Comment (NkFab) [inline server/controllers/article.controller.js]: Why does this status `200` differ from the one returned in the body `204`?

File: server/routes/article.route.js [hunk 2]
```diff
@@ -4,6 +4,7 @@ import json from 'express';
 import Article from '../controllers/article.controller';
 import verifyUser from '../middleware/authentication.middleware';
 import verifyAuthor from '../middleware/authorization.middleware';
+// import verifyArticle from '../middleware/article.middleware';
```
Comment (NkFab) [inline server/routes/article.route.js]: Remove unnecessary commented lines.

File: server/test/app.test.js [hunk 3]
```diff
@@ -1,39 +1,39 @@
-/* eslint-disable no-undef */
-import chai from 'chai';
-import chaiHttp from 'chai-http';
-import app from '../../app';
+// /* eslint-disable no-undef */
+// import chai from 'chai';
+// import chaiHttp from 'chai-http';
+// import app from '../../app';
```
Comment (NkFab) [inline server/test/app.test.js]: Remove these unnecessary commented lines.

File: server/test/app.test.js [hunk 4]
```diff
@@ -1,39 +1,39 @@
...
 
-const { expect } = chai;
+// const { expect } = chai;
```
Comment (NkFab) [inline server/test/app.test.js]: Remove these unnecessary commented lines.

File: server/test/app.test.js [hunk 5]
```diff
@@ -1,39 +1,39 @@
...
 
-chai.use(chaiHttp);
+// chai.use(chaiHttp);
```
Comment (NkFab) [inline server/test/app.test.js]: Remove these unnecessary commented lines.

File: server/test/app.test.js [hunk 6]
```diff
@@ -1,39 +1,39 @@
...
 
-describe('Teamwork API root endpoint', () => {
-  it('Does the application exist?', () => {
-    expect(app).to.be.a('function');
-  });
+// describe('Teamwork API root endpoint', () => {
+//   it('Does the application exist?', () => {
+//     expect(app).to.be.a('function');
+//   });
```
Comment (NkFab) [inline server/test/app.test.js]: Remove these unnecessary commented lines.

File: server/test/app.test.js [hunk 7]
```diff
@@ -1,39 +1,39 @@
...
+//       .get('/')
+//       .then(res => {
+//         expect(res).to.have.status(200);
+//         expect(res.body.message).to.be.equal('Teamwork API');
+//         done();
+//       })
+//       .catch(err => {
+//         return err.message;
+//       });
+//   });
```
Comment (NkFab) [inline server/test/app.test.js]: Remove these unnecessary commented lines.

File: server/test/app.test.js [hunk 8]
```diff
@@ -1,39 +1,39 @@
...
+//     chai
+//       .request(app)
+//       .get('/random/url')
+//       .end((err, res) => {
+//         expect(res.body.status).to.be.equal(405);
+//         expect(res.body.message).to.be.equal('Method not allowed');
+//         done();
+//       });
+//   });
+// });
```
Comment (NkFab) [inline server/test/app.test.js]: Remove these unnecessary commented lines.

File: server/test/article.test.js [hunk 9]
```diff
@@ -2,148 +2,168 @@
...
+  updatedArticle,
+  newComment
+} from './article.mock';
 
 const { expect } = chai;
 
 chai.use(chaiHttp);
 
+let employeeToken;
+// let userId;
```
Comment (NkFab) [inline server/test/article.test.js]: Remove these unnecessary commented lines.

File: server/controllers/article.controller.js (hunk 1 above)
Comment (sebalu) [reply server/controllers/article.controller.js]: A ```204``` on the res object returns a null response. It makes it hard for a Client that may be expecting an explicit deletion confirmation response to act accordingly. I'll refactor both to a ```200``` to be consistent.

**Pull request merged**